### ISSUE TITLE
Remote write configuration not propagated unless prometheus version is explicitly specified

### ISSUE THREAD

**Issue by jmichalek132** [prometheus-operator/prometheus-operator#3950]
**What happened?**

Remote write headers did not get passed to prometheus. I put them in prometheus spec manifest but they did not get propagated into prometheus configuration. After looking at the code related to this I realized that this is caused by logic which checks if I have prometheus version which supports this feature.

```
		if len(spec.Headers) > 0 && version.GTE(semver.MustParse("2.25.0")) {
			cfg = append(cfg, yaml.MapItem{Key: "headers", Value: stringMapToMapSlice(spec.Headers)})
		}
```
https://github.com/prometheus-operator/prometheus-operator/blob/7f94a06b86d41c20176f0d5b53aa0100fdc361e6/pkg/prometheus/promcfg.go#L1404

We don't specify `version` in prometheus spec since it not required but after bit more digging I realized that if I don't explicitly specify it falls back on `PrometheusCompatibilityMatrix` which doesn't contain the `2.25.0` version.
https://github.com/prometheus-operator/prometheus-operator/blob/v0.46.0/pkg/operator/defaults.go
  
This essentially disable feature because field which is not required is not filled in. I think this is quite bad since afaik it's not documented, it also makes wonder where else is this logic reused. I would suggest updating at least the description for remoteWrite headers. 

**Did you expect to see something different?**

Since I was running prometheus version which supports this feature I was expecting them to get propagated to the prometheus configuration.

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**

* Prometheus Operator version:

    `v0.46.0`
    

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}

Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.6-eks-49a6c0", GitCommit:"49a6c0bf091506e7bafcdb1b142351b69363355a", GitTreeState:"clean", BuildDate:"2020-12-23T22:10:21Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```
    

* Kubernetes cluster kind:

    GKE

* Manifests:

```
  remoteWrite:
  - headers:
      THANOS-TENANT: dev1
    name: thanos-receive
    tlsConfig:
      caFile: /tls/ca.pem
      certFile: /tls/prometheus.crt
      keyFile: /tls/prometheus.key
    url: https://ednpoint/api/v1/receive
```

* Prometheus Operator Logs:

```
I don't have logs for this unfortunately 
```

**Anything else we need to know?**:

**Comment by simonpasquier** [comment]
Right it was an oversight when we released the latest version of the operator (we should have bumped to the default version of Prometheus). It would be fixed by #3946 and v0.47 is planned for next week.

**Comment by jmichalek132** [comment]
Okay thank you.
